Sources/Networking/AlamofireNetworkClient/AlamofireNetworkClient.swift

```diff
@@ -253,12 +253,31 @@ public extension AlamofireNetworkClient.Error {
 // MARK: - Request API
 public extension AlamofireNetworkClient.Request {
   var asData: Promise<Data> {
+    asDataWithHeaders
+      .map { $0.0 }
+  }
+  
+  var asVoid: Promise<()> {
+    asVoidWithHeaders
+      .asVoid
+  }
+  
+  func decode<D: Decodable>(
+    _ decodable: D.Type,
+    decoder: JSONDecoder = .init(),
+    on dispatchQueue: DispatchQueue? = .main
+  ) -> Promise<D> {
+    decodeWithHeaders(decodable, decoder: decoder, on: dispatchQueue)
+      .map { $0.0 }
+  }
+  
+  var asDataWithHeaders: Promise<(Data, HTTPHeaders?)> {
```

Comment:
Instead of introducing new `*withHeaders` methods, should we rather always return headers instead?
e.g.
```swift
var asData: Promise<(Data, Headers?)>
var asVoid: Promise<(Headers?)>
```

Reply:
This would break all the existing code.

Reply:
You are right. Since this is really and edge case, we will probably never use it. And to tie it to existing code, doesn't make much sense.